src/core/sorted_map.cc

```diff
@@ -356,6 +371,33 @@ bool SortedMap::RdImpl::Iterate(unsigned start_rank, unsigned len, bool reverse,
   return success;
 }
 
+uint64_t SortedMap::RdImpl::Scan(uint64_t cursor,
+                                 absl::FunctionRef<void(std::string_view, double)> cb) const {
+  auto scanCb = [](void* privdata, const dictEntry* de) {
+    auto* cb = (absl::FunctionRef<void(std::string_view, double)>*)privdata;
+
+    sds key = (sds)de->key;
+    double score = *(double*)dictGetVal(de);
+    (*cb)(std::string_view(key, sdslen(key)), score);
+  };
```

Comment:
Got it, it decays to a function pointer below only if it doesn't capture anything, nit reinterpret_cast is by the styleguide